rs/index/src/posting_list/file.rs

```diff
@@ -254,30 +256,36 @@ impl<'a> PostingListStorage<'a> for FileBackedAppendablePostingListStorage {
             )]));
         }
 
-        if i >= self.current_num_of_posting_list {
+        if i >= self.entry_count {
             return Err(anyhow!("Posting list id out of bound"));
         }
         let size_in_bytes = std::mem::size_of::<u64>();
```

Comment:
use a different name a well. whose size is this? otherwise, just inline the variable

Reply:
I feel like the logic to extract information from both metadata and posting list files are the same. Maybe there is a way to reuse the code here?
* Get the overall offset
* From the overall offset, get the file acess info
* From the file access info, given the length you need to read, read until finish